tpmtool.cpp

```diff
@@ -626,121 +632,215 @@ CreateSpace (
-           "R" : "NA");
-    printf("Auth Rights  : %s\n",
-           ((authRights & TpmToolReadWriteAccess) == TpmToolReadWriteAccess) ?
-           "RW" :
-           ((authRights & TpmToolReadAccess) == TpmToolReadAccess) ?
-           "R" : "NA");
+    if ((ownerRights & TpmToolReadWriteAccess) == TpmToolReadWriteAccess)
+    {
+        strcpy(OwnerRights, "RW");
+    }
+    else if((ownerRights & TpmToolReadAccess) == TpmToolReadAccess)
+    {
+        strcpy(OwnerRights, "R");
+    }
+    else
+    {
+        strcpy(OwnerRights, "NA");
+    }
 
-    //
-    // Do the attributes, first with the ones set at creation
-    //
-    printf("Attributes   :");
-    strcpy_s(attributeBuffer, sizeof(attributeBuffer), " ");
+    if ((authRights & TpmToolReadWriteAccess) == TpmToolReadWriteAccess)
+    {
+        strcpy(AuthRights, "RW");
+    }
+    else if ((authRights & TpmToolReadAccess) == TpmToolReadAccess)
+    {
+        strcpy(AuthRights, "R");
+    }
+    else
+    {
+        strcpy(AuthRights, "NA");
+    }
+
+    strcpy(AttributeBuffer, " ");
     if (attributes & TpmToolReadLockable)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "RL+");
+        strcat(AttributeBuffer, "RL+");
     }
     if (attributes & TpmToolWriteLockable)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "WL+");
+        strcat(AttributeBuffer, "WL+");
     }
     if (attributes & TpmToolWriteOnce)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "WO+");
+        strcat(AttributeBuffer, "WO+");
     }
     if (attributes & TpmToolWriteAll)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "WA+");
+        strcat(AttributeBuffer, "WA+");
     }
     if (attributes & TpmToolNonProtected)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "NP+");
+        strcat(AttributeBuffer, "NP+");
     }
     if (attributes & TpmToolCached)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "CH+");
+        strcat(AttributeBuffer, "CH+");
     }
     if (attributes & TpmToolVolatileDirtyFlag)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "VL+");
+        strcat(AttributeBuffer, "VL+");
     }
     if (attributes & TpmToolPermanent)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "PT+");
+        strcat(AttributeBuffer, "PT+");
     }
 
     //
     // Next, the status attributes
     //
     if (attributes & TpmToolReadLocked)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "LR+");
+        strcat(AttributeBuffer, "LR+");
     }
     if (attributes & TpmToolWriteLocked)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "LW+");
+        strcat(AttributeBuffer, "LW+");
     }
     if (attributes & TpmToolPlatformOwned)
     {
-        strcat_s(attributeBuffer, sizeof(attributeBuffer), "PO+");
+        strcat(AttributeBuffer, "PO+");
     }
+    AttributeBuffer[strlen(AttributeBuffer) - 1] = ' ';
```

Comment:
I don't think this is needed... my hack converts the + to a \n already, lower down.

Reply:
And in your code, it gets converted into a 0, so i don't see why we need the space?